stdlib/public/core/Dictionary.swift

```diff
@@ -785,6 +785,60 @@ extension Dictionary {
         removeValue(forKey: key)
       }
     }
+    _modify {
+      // FIXME: This code should be moved to _variant, with Dictionary.subscript
+      // yielding `&_variant[key]`.
+
+      // Look up (empty or occupied) bucket corresponding to the given key.
+      let isUnique = _variant.isUniquelyReferenced()
+      var idealBucket = _variant.asNative._bucket(key)
+      var (pos, found) = _variant.asNative._find(key, startBucket: idealBucket)
+
+      // Prepare storage.
+      // If `key` isn't in the dictionary yet, assume that this access will end
+      // up inserting it. Otherwise this can only be a removal or an in-place
+      // mutation. (If we guess wrong, we might needlessly rehash; that's fine.)
+      let (_, rehashed) = _variant.ensureUniqueNative(
+        withCapacity: self.capacity + (found ? 0 : 1),
```

Comment:
Good thing benchmarks don’t exercise this yet — this should use self.count, not capacity. (As it stands, this forces a full 2x storage reallocation & rehash on every insertion.) 😳